coc/client.py

```diff
@@ -278,14 +278,20 @@ def login_with_keys(self, *keys: str) -> None:
 
         LOG.debug("HTTP connection created. Client is ready for use.")
 
-    def close(self) -> None:
+    def close_loops(self) -> None:
```

Comment:
Only little thing I'm worried about is making sure we don't break stuff in a non-major version, ie. I think (please correct me if I'm wrong) but client.close() now doesn't exist. 

Long-term, should we always be creating and closing clients with await stuff and letting asyncio handle running async stuff in non-async places? I wonder whether that would solve all our issues, and whether I'm trying to do something that isn't really my job.

So change `.close()` to be an async function and either make `coc.login()` an async function or remove it completely.

Like you showed, it's easier (and better if it doesn't have all those loop cleanup issues) to leave it to asyncio.run() to run it all. Thoughts? I'm kinda just rambling. Either way, I think we can add this stuff with different names now, but need to keep the existing functions until v3 (maybe adding a depreciated warning)

Reply:
asyncio.run() requieres python 3.7 or newer, we would need to add that somewhere too.

The big downside with asyncio.run() is most likely the usage of a coc client with another aiohttp session, like a discord bot. We should definitely test that before we discussing this solution further.

Reply:
Absolutely, and very true re. python 3.7

I guess the `asyncio.run` stuff would only be for people running it as a standalone script thing - else the current `coc.login` is fine (and doesn't have the loop cleanup issues / people don't care as much because there's a hundred other warnings when you close a bot). I think the real issue is `client.close()`, not login so much. I don't think people generally have ever had many issues with login in it's current form

Reply:
So you actually mentioned it I personally do not think it is on your library to handle the cleanup because it will create a race condition with other libraries that also depend on sessions to be cleaned out. 

This is the best way I have found to handle cleanup of both coc.py sessions and asyncpg and that's to drop into "manual" loops instead of the abstracted run only to be able to properly clean up after myself. The asyncio.run does do the generator cleanup but it will not properly clean up sessions. Which may or may not be an issue. 

https://github.com/majordoobie/PantherLily/blob/main/main.py#L117